test/functional/p2p_net_deadlock.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python3
+# Copyright (c) 2023-The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+import threading
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import random_bytes
+
+
+def send_message(node, peer_id, msg_type, msg):
+    node.sendmsgtopeer(peer_id, msg_type, msg)
+
+
+class NetDeadlockTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+
+    def setup_network(self):
+        self.setup_nodes()
+
+    def run_test(self):
+        node0 = self.nodes[0]
+        node1 = self.nodes[1]
+        self.connect_nodes(0, 1)
```

Comment:
I think you don't need to override `setup_network`:
```diff
diff --git a/test/functional/p2p_net_deadlock.py b/test/functional/p
2p_net_deadlock.py
index 1b2139cc8..4ed9170e4 100755
--- a/test/functional/p2p_net_deadlock.py
+++ b/test/functional/p2p_net_deadlock.py
@@ -17,14 +17,9 @@ class NetDeadlockTest(BitcoinTestFramework):
         self.setup_clean_chain = True
         self.num_nodes = 2
 
-    def setup_network(self):
-        self.setup_nodes()
-
     def run_test(self):
         node0 = self.nodes[0]
         node1 = self.nodes[1]
-        self.connect_nodes(0, 1)
-        self.sync_all()
 
         self.log.info("Simultaneously send a large message on both sides")
         rand_msg = random_bytes(4000000).hex()
```

Reply:
done